xarray_jax/__init__.py

```diff
@@ -719,19 +722,18 @@ def dims_change_on_unflatten(dims_change_fn: DimsChangeFn):
 
 
 def _flatten_variable(v: xarray.Variable) -> Tuple[
-    Tuple[jax.typing.ArrayLike], Tuple[Tuple[Hashable, ...], Tuple[int, ...]]]:  # pylint: disable=g-one-element-tuple
+    Tuple[jax.typing.ArrayLike], Tuple[Hashable, ...]]:  # pylint: disable=g-one-element-tuple
   """Flattens a Variable for jax.tree_util."""
   children = (unwrap_data(v),)
-  aux = (v.dims, v.shape)
+  aux = v.dims
   return children, aux
 
 
 def _unflatten_variable(
-    aux: Tuple[Tuple[Hashable, ...], Tuple[int, ...]],
-    children: Tuple[jax.typing.ArrayLike],
-) -> xarray.Variable:
+    aux: Tuple[Hashable, ...],
+    children: Tuple[jax.typing.ArrayLike]) -> xarray.Variable:  # pylint: disable=g-one-element-tuple
```

Comment:
Just noticed this should now be `children: Tuple[Any]` as we no longer assume `jax.typing.ArrayLike` for the leaf values.

Likewise in the return type of _flatten_variable above.